### ISSUE TITLE
Retroplayer is dropping frames every 10 seconds

### ISSUE THREAD

**Issue by chcore** [xbmc/xbmc#19622]
### Describe the bug
Here is a clear and concise description of what the problem is:

While playing a SNES game with any of the available cores, Retroplayer will drop a bunch of frames every 10 seconds.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

The gameplay should be perfectly smooth at all times. No frames should be dropped on hardware that greatly exceeds what is required to emulate SNES games.

## Actual Behavior

Frame drops every 10 seconds.

### To Reproduce

1. Install Kodi
2. Play a SNES game, preferably a side scroller with smooth motion (Super Mario Bros. is a good one)
3. Notice the frame drops every 10 seconds

### Debuglog

The debuglog can be found here:

http://ix.io/2U6j

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

A video that clearly shows the issue at 16s and 26s: https://www3.zippyshare.com/v/ukRSx2e7/file.html

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Arch Linux, LibreELEC and Windows 10 (the issue exists on all of them)
 - Kodi version: 19

The issue occurrs on my Tiger Lake NUC as well as an Intel/Nvidia desktop. Disabling the save/rewind features does not fix the issue, nor does disabling any CPU/GPU throttling settings via the BIOS or OS. I play NTSC ROMs with the refresh rate at 60Hz.

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by lrusak** [comment]
Initial reaction tells me something is wrong with your audio device. There seems to be a bit of spew at the start of the log that I've never seen before. I see you are using the output device `MEI Panasonic-TV on DisplayPort #5` are you actually using displayport or is this some internal conversion (LSPCON)? Can you try with another port?

Can you try at another refresh rate and/or resolution to see if it happens when not using 1080p60?

Otherwise I don't have any ideas as the log doesn't show me anything wrong when playing the game.

**Comment by chcore** [comment]
Yes it's actually a HDMI port. I have already tried a HDMI ports, and even a different display entirely (a PC monitor).

I don't think it has anything to do with the audio device, as I have already tried analogue output. I have also experienced the same FPS drops on a completely separate desktop machine (the Intel/Nvidia/Windows 10 machine mentioned above).

I have tried 720p60, 1080p60 and 2160p60, it does not make a difference. My TV also supports 59Hz, it's no better.

Running standalone Snes9x on Windows 10 is completely smooth. As is running standalone Retroarch on linux.

**Comment by smp79** [comment]
I can reproduce this issue as well. It is noticeable only in side scrolling games like Super Mario Bros. There are definitely micro-stutters every few seconds of gameplay. I tested on different machines with LibreELEC and Windows 10.

**Comment by pasikarkkainen** [comment]
Probably related to framerate conversion? SNES refresh rate is  60.098 Hz, and if kodi/retroplayer is outputting at 60.00 Hz you'll definitely get stuttering every x seconds.

**Comment by chcore** [comment]
> Probably related to framerate conversion? SNES refresh rate is 60.098 Hz, and if kodi/retroplayer is outputting at 60.00 Hz you'll definitely get stuttering every x seconds.

Sounds plausible. I don't know how other emulators handle this though.

**Comment by smvicente** [comment]
I can confirm this also happens to me. Tried all cores for SNES and it indeed seems to be refresh rate related. With other machines emulators, like MAME, it doesn't happen.

**Comment by smvicente** [comment]
In RetroArch there is an option called "Sync to exact content framerate", which should be disabled unless using a Freesync/Gsync monitor, just in case it helps for troubleshooting when comparing to Retroplayer.

**Comment by garbear** [comment]
Fixed by https://github.com/xbmc/xbmc/pull/29227. Caused by the synchronous nature of a bad decision I made in 2016 to confirm RetroPlayer to the same API as VideoPlayer, which is unneeded and never should have occurred. After the merge, the 10s stutters should go away for all cores.
